plugins/wandb_plugin.py

```diff
@@ -14,11 +16,31 @@ class WandbConfig:
     project_id: str
 
 
+class WandbCallback(TrainerCallback):
+    def __init__(self, wandb):
+        self.wandb = wandb
+
+    def on_log(self, args, state, control, logs=None, **kwargs):
```

Comment:
hmm it doesn't look like this could be generalizable to any other model. what would we pass SKLearn or for pytorch pass in for "state" and "control"? I think this should be a more standard interface, like `log(logs)`, and TrainerCallback should be wrapping our more generic `log` function inside a Huggingface model! So we can pass around a more general `log` function instead of a HF-specific class that no other model could use.